### PR TITLE
[Core] Extend B&S test with MPC

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**

Extend B&S test with MPC

**🆕 Changelog**

- [Extend B&S test with MPC](https://github.com/KratosMultiphysics/Kratos/commit/1846bccfa89153a142e37aa6cea5503acb2a5ca8)

### CODE REVIEW

File: kratos/tests/cpp_tests/strategies/builder_and_solvers/test_builder_and_solver.cpp [hunk 1]
```diff
@@ -502,6 +502,16 @@ namespace Kratos::Testing
...
         KRATOS_CHECK_RELATIVE_NEAR(rA(5,5), 1.0, tolerance);
+
+        const auto& r_T = p_builder_and_solver->GetConstraintRelationMatrix();
+        KRATOS_CHECK(r_T.size1() == 6);
+        KRATOS_CHECK(r_T.size2() == 6);
+        KRATOS_CHECK_RELATIVE_NEAR(r_T(0,0), 1.0, tolerance);
+        KRATOS_CHECK_RELATIVE_NEAR(r_T(1,1), 1.0, tolerance);
+        KRATOS_CHECK_RELATIVE_NEAR(r_T(2,2), 1.0, tolerance);
+        KRATOS_CHECK_RELATIVE_NEAR(r_T(3,3), 1.0, tolerance);
+        KRATOS_CHECK_RELATIVE_NEAR(r_T(4,2), 1.0, tolerance);
```
Comment (matekelemen) [inline kratos/tests/cpp_tests/strategies/builder_and_solvers/test_builder_and_solver.cpp:513]: `4,4` perhaps, or am I missing something? If it is, a loop would be more compact.

Comment (loumalouomega) [reply kratos/tests/cpp_tests/strategies/builder_and_solvers/test_builder_and_solver.cpp:513]: Nope, it is the T matrix, and we are imposing node 5 behaves as node 3, that's why

Comment (loumalouomega) [reply kratos/tests/cpp_tests/strategies/builder_and_solvers/test_builder_and_solver.cpp:513]: It's the point of the test

Comment (matekelemen) [reply kratos/tests/cpp_tests/strategies/builder_and_solvers/test_builder_and_solver.cpp:513]: ah, alright then

Comment (loumalouomega) [reply kratos/tests/cpp_tests/strategies/builder_and_solvers/test_builder_and_solver.cpp:513]: Approval?

**Review by roigcarlo — APPROVED** [review]

**Comment by loumalouomega** [comment]
Thanks @roigcarlo

**Pull request merged**